File: shared/paths.py
import os

from pathlib import Path
from dotenv import dotenv_values

fp = Path(__file__)
src_path = fp.resolve().parents[1]
cfg = dotenv_values(src_path.joinpath(".env"))

class Paths:
    """
    Project paths for easy reference.
    """
    project_path = src_path.parents[0]
    src_path = src_path
    common = os.path.join(src_path, 'shared')

    path_trade = Path(cfg['PATH_TRADE'])
    analytics = path_trade.joinpath('Analytics')
    path_models = path_trade.joinpath('models')
    path_data = path_trade.joinpath('data')
    path_ib = path_trade.joinpath('ib')

    path_data_alternative = path_data.joinpath('alternative')
    path_data_interest_rate = path_data_alternative.joinpath('interest-rate')
    path_symbol_properties = path_data.joinpath('symbol-properties')
    path_market_hours = path_data.joinpath('market-hours')
    path_activity_reports_ytd = path_ib.joinpath('activityReportsYTD')
    path_analysis_frames = analytics.joinpath('analysis_frames')
    path_calibration = analytics.joinpath('calibration')
    path_api_cache = analytics.joinpath('api_cache')

    path_earnings = path_symbol_properties.joinpath('EarningsAnnouncements.json')
    path_dividend_yields = path_symbol_properties.joinpath('DividendYields.json')
    path_market_hours_database = path_market_hours.joinpath('market-hours-database.json')


def mkdir(path: str | Path) -> str | Path:
    if not os.path.exists(path):
        os.makedirs(path)
    return path

File: fino/src/shared/Paths.jl
module Paths

using Dates

const LOG_FN = "log_{}.txt"
const SRC_PATH = joinpath(@__DIR__, "..")
const PROJECT_PATH = joinpath(SRC_PATH, "..")
const COMMON = joinpath(SRC_PATH, "shared")

# Environment variable (assumes PATH_TRADE is set; use get(ENV, "PATH_TRADE", default) if optional)
const PATH_TRADE = ENV["PATH_TRADE"]
const ANALYTICS = joinpath(PATH_TRADE, "Analytics")
const PATH_MODELS = joinpath(PATH_TRADE, "models")
const PATH_DATA = joinpath(PATH_TRADE, "data")
const PATH_IB = joinpath(PATH_TRADE, "ib")

const PATH_DATA_ALTERNATIVE = joinpath(PATH_DATA, "alternative")
const PATH_DATA_INTEREST_RATE = joinpath(PATH_DATA_ALTERNATIVE, "interest-rate")
const PATH_SYMBOL_PROPERTIES = joinpath(PATH_DATA, "symbol-properties")
const PATH_MARKET_HOURS = joinpath(PATH_DATA, "market-hours")
const PATH_ACTIVITY_REPORTS_YTD = joinpath(PATH_IB, "activityReportsYTD")
const PATH_ANALYSIS_FRAMES = joinpath(ANALYTICS, "analysis_frames")
const PATH_CALIBRATION = joinpath(ANALYTICS, "calibration")
const PATH_API_CACHE = joinpath(ANALYTICS, "api_cache")

const PATH_EARNINGS = joinpath(PATH_SYMBOL_PROPERTIES, "EarningsAnnouncements.json")
const PATH_DIVIDEND_YIELDS = joinpath(PATH_SYMBOL_PROPERTIES, "DividendYields.json")
const PATH_MARKET_HOURS_DATABASE = joinpath(PATH_MARKET_HOURS, "market-hours-database.json")

export LOG_FN, SRC_PATH, PROJECT_PATH, COMMON, PATH_TRADE, ANALYTICS,
PATH_MODELS, PATH_DATA, PATH_IB, PATH_DATA_ALTERNATIVE,
PATH_DATA_INTEREST_RATE, PATH_SYMBOL_PROPERTIES, PATH_MARKET_HOURS,
PATH_ACTIVITY_REPORTS_YTD, PATH_ANALYSIS_FRAMES, PATH_CALIBRATION,
PATH_API_CACHE, PATH_EARNINGS, PATH_DIVIDEND_YIELDS,
PATH_MARKET_HOURS_DATABASE, mkdir

"""
Create directories if they don't exist. Returns the path.
"""
function mkdir(path::AbstractString)
    mkpath(path)
    return path
end

end # module Paths
